src/NHibernate/Linq/ExpressionTransformers/RemoveRedundantCast.cs

```diff
@@ -20,6 +21,7 @@ public class RemoveRedundantCast : IExpressionTransformer<UnaryExpression>
 		public Expression Transform(UnaryExpression expression)
 		{
 			if (expression.Type != typeof(object) &&
+				expression.Type != typeof(Enum) &&
```

Comment:
This part is fix for exception thrown in SelectChildClassContainedInParent test.